Messaging.md

```diff
@@ -11,55 +11,88 @@ All data fields are unsigned big-endian unless otherwise specified.
-      - [Version 0 `enum_event_descriptor`](#version-0-enum_event_descriptor)
-      - [Version 0 `digit_decomposition_event_descriptor`](#version-0-digit_decomposition_event_descriptor)
+      - [`enum_event_descriptor`](#enum_event_descriptor)
+      - [`digit_decomposition_event_descriptor`](#digit_decomposition_event_descriptor)
    - [The `oracle_event` Type](#the-oracle_event-type)
-      - [Version 0 `oracle_event`](#version-0-oracle_event)
+      - [`oracle_event`](#oracle_event)
    - [The `oracle_announcement` Type](#the-oracle_announcement-type)
-      - [Version 0 `oracle_announcement`](#version-0-oracle_announcement)
+      - [`oracle_announcement`](#oracle_announcement)
    - [The `oracle_attestation` Type](#the-oracle_attestation-type)
-      - [Version 0 `oracle_attestation`](#version-0-oracle_attestation)
+      - [`oracle_attestation`](#oracle_attestation)
 - [Authors](#authors)
 
-
 ## Connection Handling and Multiplexing
 
 Implementations MUST use a single connection per peer; contract messages (which include a contract ID) are multiplexed over this single connection.
 
 ## Message Format
 
-We reuse the [Lightning Message Format](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#lightning-message-format) and the [Type-Length-Value Format](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#type-length-value-format) (TLV).
-To be clear, any encoded binary blob that can be sent over the wire will follow the Lightning Message Format
-while all sub-types internal to these messages will follow the Type-Length-Value Format.
-This means that types on outer-messages will be represented with `u16` integers (defined below) and their length
-is omitted from their encoding because the transport layer has the length in a separate unencrypted field.
-Meanwhile all typed sub-messages (which follow TLV format) will have their types represented using `bigsize` integers
-(defined below) and their lengths (also `bigsize`) are included in their encodings.
+We reuse the [Lightning Message Format](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#lightning-message-format) for messages sent over the wire, the [Type-Length-Value Format](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#type-length-value-format) (TLV) for message extensibility and extra features support, and a custom format for [sub-types](#sub-types).
+
+### Wire Messages
+
+Any encoded binary blob that can be sent over the wire will follow the Lightning Message Format.
+This means that wire messages are prefixed with a `u16` field (defined below) and their length is omitted from their encoding because the transport layer has the length in a separate un-encrypted field.
+
+### Type-Length-Value
+
+A TLV stream is defined for each wire message, and can be used to extend the protocol and for application specific features.
```

Comment:
Browsing through the BOLTs again, `tlv_stream` are extensions to already defined messages, allowing to add optional fields in a backward compatible-way ?

```
Throughout the protocol, a TLV (Type-Length-Value) format is used to allow for the backwards-compatible addition of new fields to existing message types.
```

IIUC your proposal correctly it defines the content of any of our DLC messages as a TLV namespace itself ? Not sure if it's achieving it's purpose of bandwidth-saving as the type/length components of a `tlv_record` are redundant information for mandatory fields ?

Reply:
My point was to have `tlv_stream` for the exact same purpose as LN, so maybe the way I've written things is confusing?